### ISSUE TITLE
Decoding error for int256[]

### ISSUE THREAD

**Issue by Georgi87** [ethereum/web3.js#211]
Using the latest geth version and the latest web3 version (master and develop), I get an error while decoding the response of one of my contract functions.

geth:
```
Geth
Version: 0.9.22
Protocol Version: 60
Network Id: 0
Go Version: go1.4.2
OS: linux
GOPATH=
GOROOT=/usr/lib/go
```

The abi of my function is:

```
{
    "name": "get_market_hashes(int256)",
    "type": "function",
    "inputs": [{ "name": "event_hash", "type": "int256" }],
    "outputs": [{ "name": "out", "type": "int256[]" }]
}
```

The function execution results in:
```
// arguments
["0x39ab5fd77e63aa7f25f74952405fcd2a"]
// payload
function.js:102 Object {to: "0xee636d822d52524bbb0decae737b2504cfb09760", data: "0xbae156b70000000000000000000000000000000039ab5fd77e63aa7f25f74952405fcd2a"}
function.js:105 
// response
0x00000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000000
// error
angular.js:11607 BigNumber Error: new BigNumber() not a base 16 number: 
    at raise (http://192.168.0.101:8050/static/dapp/lib-js/web3.js:6432:25)
    at http://192.168.0.101:8050/static/dapp/lib-js/web3.js:6420:33
    at new BigNumber (http://192.168.0.101:8050/static/dapp/lib-js/web3.js:5467:28)
    at SolidityType.formatOutput (http://192.168.0.101:8050/static/dapp/lib-js/web3.js:99:22)
    at http://192.168.0.101:8050/static/dapp/lib-js/web3.js:219:29
    at Array.map (native)
    at SolidityCoder.decodeParams (http://192.168.0.101:8050/static/dapp/lib-js/web3.js:216:18)
    at SolidityFunction.unpackOutput (http://192.168.0.101:8050/static/dapp/lib-js/web3.js:2754:24)
    at SolidityFunction.call (http://192.168.0.101:8050/static/dapp/lib-js/web3.js:2777:21)
    at updateMarkets (http://192.168.0.101:8050/static/dapp/js/services.js:183:75)
```

**Comment by ethers** [comment]
same as #210 ;)

**Comment by debris** [comment]
fixed in 0.4.3
